packages/react-native-payments/lib/js/PKPaymentButton.js

```diff
@@ -46,12 +48,32 @@ export class PKPaymentButton extends React.Component<Props> {
     height: 44,
   };
 
+  state = {
+    defaultToSetup: false
+  }
+
+  async componentDidMount() {
+    const { supportedNetworks: customNetworks } = this.props;
+    let supportedNetworks = customNetworks;
+    if (supportedNetworks.length === 0) {
+      supportedNetworks = await PaymentRequest.availableNetworks();
+    }
+    const havePaymentSetup =
+      await PaymentRequest.canMakePaymentsUsingNetworks(supportedNetworks);
+    if (!havePaymentSetup) {
+      this.setState({ defaultToSetup: true })
+    }
+  }
+
   render() {
+    const { defaultToSetup } = this.state;
+    const buttonType = defaultToSetup ? 'setUp' : this.props.buttonType;
+    const onPress = defaultToSetup ? PaymentRequest.openPaymentSetup : this.props.onPress;
     return (
       <RNPKPaymentButton
         buttonStyle={this.props.style}
-        buttonType={this.props.type}
```

Comment:
@chukohsin 

Since `type` isn't being used anymore, let's just remove it from the props then:
https://github.com/brandingbrand/react-native-payments/pull/18/files#diff-580f7fb11446db36a63d29434c9cb92eR29

App-side shouldn't be deciding this button type anyways.

This might break props for some apps though, so mark this as a breaking change then.

Reply:
@CuriousMark Thanks for this. Found a type on line 70. It should be `this.props.type` instead of `this.props.buttonType`. Will fix it!